autolabel/source/filetype_checker.py

```diff
@@ -0,0 +1,94 @@
+"""
+This Python script contains functions to check various types of files and paths.
+
+Usage:
+You can import these functions into your script and use them to check different types of files and paths. Additionally, you can add test code within the if __name__ == "__main__" block to validate the functionality of these functions.
+"""
+
+import os
+import imghdr
+import fnmatch
+import mimetypes
+
+def is_file(src: str) -> bool:
+    """
+    Check if the given path is a file.
+    """
+    return os.path.isfile(src)
+
+def is_stream(src: str) -> bool:
+    """
+    Check if the given path represents a stream.
+    """
+    try:
+        with open(src, 'rb') as file:
+            return hasattr(file, 'readable') and hasattr(file, 'seekable')
+    except IOError:
+        return False
+
+def is_screenshot(src: str) -> bool:
```

Comment:
imghdr will remove from python 3.13. and we just use string type to determine the input source. like 'screen:0'